src/utils/getBreadcrumbFromSiteMap.ts

```diff
@@ -0,0 +1,33 @@
+import type { BreadcrumbProps } from "~/common"
+import type { IsomerSitemap } from "~/engine"
+
+// Traverse the sitemap to get the breadcrumb for the page with the given
+// permalink
+export const getBreadcrumbFromSiteMap = (
+  sitemap: IsomerSitemap,
+  permalink: string[],
+): BreadcrumbProps => {
+  const breadcrumb = []
+  let node = sitemap
+  let currentPath = ""
+
+  for (const pathSegment of permalink) {
+    currentPath += "/" + pathSegment
+    const nextNode = node.children?.find(
+      (node) => node.permalink === currentPath,
+    )
+
+    if (!nextNode) {
+      // TODO: handle this unexpected case where cannot traverse to permalink in the sitemap
```

Comment:
hmm why not just throw / log for now so no silent failures ah?

Reply:
Both the permalink and the sitemap are autogenerated from the same data source, which is the folder structure of the site repository stored on the file system, so this case shouldn't happen. Opted to deprioritise this until we minimally achieve feature parity.

Reply:
We currently don't have a logging mechanism setup yet, so we might not be able to do this right now. If we do a `console.log` here it will print out to the browser's console which would be seen by site visitors.